### PR TITLE
Sptd 577

### PR DESCRIPTION
**Pull request by zackschmidt**
## Related Issue(s)

[SPTD-577](https://cdw.atlassian.net/browse/SPTD-577)
## Description

Fix a bug where best-practices/add-rolebinding/add-rolebinding.yaml is missing apiVersion, which throws the error
E0706 14:25:58.262143 903007 apply_command.go:391] "msg"="policy validation error" "error"="path: spec.rules[0].generate.apiVersion.: apiVersion cannot be empty"

Restrict image registries for best-practices/restrict-image-registries/restrict-image-registries.yaml to be the ones pulled from our k8s production cluster. 

Getting all passes for 
`zacschm@LT-A0093677 ~/dev/mans-kyverno-policies/best-practices (sptd-577)$ kyverno test .`
See story notes for details

## Checklist

- [] I have read the [policy contribution guidelines](https://github.com/kyverno/policies/blob/main/README.md#contribution).
- [] I have added test manifests and resources covering both positive and negative tests that prove this policy works as intended.
- [] I have added the artifacthub-pkg.yml file and have verified it is complete and correct.

### CODE REVIEW

**Review by LongBeachHXC — APPROVED** [review]

**Comment by zackschmidt** [comment]
@cdwbrad No reason, but I am having trouble getting my editor to modifying that

File: best-practices/require-ro-rootfs/bad-pod-false.yaml [hunk 1]
```diff
@@ -7,4 +7,4 @@ spec:
   - name: busybox
     image: busybox:1.35
     securityContext:
-      readOnlyRootFilesystem: false
```
Comment (cdwbrad) [inline best-practices/require-ro-rootfs/bad-pod-false.yaml:10]: If we're not materially changing code, we should not be submitting these sort of changes to a public repo.  It just leads to more noise when we merge from public upstream. Can this change be reverted?

File: best-practices/require-ro-rootfs/bad-pod-notall.yaml [hunk 2]
```diff
@@ -22,4 +22,4 @@ spec:
     securityContext:
       readOnlyRootFilesystem: true
   - name: busybox
-    image: busybox:1.35
\ No newline at end of file
+    image: busybox:1.35
```
Comment (cdwbrad) [inline best-practices/require-ro-rootfs/bad-pod-notall.yaml:25]: If we're not materially changing code, we should not be submitting these sort of changes to a public repo.  It just leads to more noise when we merge from public upstream. Can this change be reverted?

File: best-practices/require-ro-rootfs/bad-pod-nothing.yaml [hunk 3]
```diff
@@ -5,4 +5,4 @@ metadata:
 spec:
   containers:
   - name: busybox
-    image: busybox:1.35
\ No newline at end of file
+    image: busybox:1.35
```
Comment (cdwbrad) [inline best-practices/require-ro-rootfs/bad-pod-nothing.yaml:8]: If we're not materially changing code, we should not be submitting these sort of changes to a public repo.  It just leads to more noise when we merge from public upstream. Can this change be reverted?

File: best-practices/require-ro-rootfs/bad-podcontrollers.yaml [hunk 4]
```diff
@@ -37,4 +37,4 @@ spec:
           - name: busybox-again
             image: busybox:1.35
             securityContext:
-              readOnlyRootFilesystem: true
\ No newline at end of file
+              readOnlyRootFilesystem: true
```
Comment (cdwbrad) [inline best-practices/require-ro-rootfs/bad-podcontrollers.yaml:40]: If we're not materially changing code, we should not be submitting these sort of changes to a public repo.  It just leads to more noise when we merge from public upstream. Can this change be reverted?

File: best-practices/require-ro-rootfs/good-podcontrollers.yaml [hunk 5]
```diff
@@ -41,4 +41,4 @@ spec:
           - name: busybox-again
             image: busybox:1.35
             securityContext:
-              readOnlyRootFilesystem: true
\ No newline at end of file
+              readOnlyRootFilesystem: true
```
Comment (cdwbrad) [inline best-practices/require-ro-rootfs/good-podcontrollers.yaml:44]: If we're not materially changing code, we should not be submitting these sort of changes to a public repo.  It just leads to more noise when we merge from public upstream. Can this change be reverted?

File: best-practices/add-rolebinding/add-rolebinding.yaml [hunk 6]
```diff
@@ -44,4 +45,4 @@ spec:
             name: admin
           subjects:
             - kind: User
-              name: "{{request.userInfo.username}}"
\ No newline at end of file
+              name: "{{request.userInfo.username}}"
```
Comment (cdwbrad) [inline best-practices/add-rolebinding/add-rolebinding.yaml:48]: If we're not materially changing code, we should not be submitting these sort of changes to a public repo.  It just leads to more noise when we merge from public upstream. Can this change be reverted?

**Review by cdwbrad — CHANGES_REQUESTED** [review]

**Review by cdwbrad — APPROVED** [review]
LGTM

**Pull request merged**